metrics/mean_iou/mean_iou.py

```diff
@@ -0,0 +1,291 @@
+        label_map (`dict`, *optional*):
+            Mapping old labels to new labels. The parameter will work only when label is str. Default: dict().
+        reduce_labels (`bool`, *optional*, defaults to `False`):
+            Whether or not to reduce all label values of segmentation maps by 1. Usually used for datasets where 0 is used for background,
+        and background itself is not included in all classes of a dataset (e.g. ADE20k). The background label will be replaced by 255.
+
+    Returns:
+        `Dict[str, float | ndarray]` comprising various elements:
+        - *mIoU* (`float`):
+            Mean Intersection-over-Union (IoU averaged over all categories).
+        - *mAcc* (`float`):
+            Mean accuracy (averaged over all categories).
+        - *overall_accuracy* (`float`):
+            Overall accuracy on all images.
+        - *per_category_accuracy* (`ndarray` of shape `(num_labels,)`):
+            Per category accuracy.
+        - *per_category_iou* (`ndarray` of shape `(num_labels,)`):
+            Per category IoU.
+    """
+    total_area_intersect, total_area_union, total_area_pred_label, total_area_label = total_intersect_and_union(
+        results, gt_seg_maps, num_labels, ignore_index, label_map, reduce_labels
+    )
+
+    # compute metrics
+    metrics = OrderedDict()
+
+    all_acc = total_area_intersect.sum() / total_area_label.sum()
+    iou = total_area_intersect / total_area_union
+    acc = total_area_intersect / total_area_label
+
+    metrics["mIoU"] = np.nanmean(iou)
+    metrics["mAcc"] = np.nanmean(acc)
+    metrics["overall_accuracy"] = all_acc
+    metrics["per_category_iou"] = iou
+    metrics["per_category_accuracy"] = acc
+
+    if nan_to_num is not None:
+        metrics = OrderedDict(
+            {metric: np.nan_to_num(metric_value, nan=nan_to_num) for metric, metric_value in metrics.items()}
+        )
+
+    return metrics
+
+
+@datasets.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
+class Mean_IoU(datasets.Metric):
+    def _info(self):
+        return datasets.MetricInfo(
+            description=_DESCRIPTION,
+            citation=_CITATION,
+            inputs_description=_KWARGS_DESCRIPTION,
+            features=datasets.Features(
+                # 1st Seq - height dim, 2nd - width dim
+                {
+                    "predictions": datasets.Sequence(datasets.Sequence(datasets.Value("uint8"))),
+                    "references": datasets.Sequence(datasets.Sequence(datasets.Value("uint8"))),
+                }
+            ),
+            reference_urls=[
+                "https://github.com/open-mmlab/mmsegmentation/blob/71c201b1813267d78764f306a297ca717827c4bf/mmseg/core/evaluation/metrics.py"
+            ],
+        )
+
+    def _compute(
+        self,
+        predictions,
+        references,
+        num_labels: int,
+        ignore_index: bool,
+        nan_to_num: Optional[int] = None,
+        label_map: Optional[
```

Comment:
Not sure if we handle this arg properly because here the default value is `None` and in `mean_iou`/`total_intersect_and_union`/`intersect_and_union` is `dict()` (I would also expect `None` there). Also, feel free to add typing to the signatures of these functions

Reply:
I'll set it to `None` everywhere.